Profiler/log/TraceLog.cpp

```diff
@@ -1,53 +1,55 @@
-#include "TraceLog.h"
-#include "version.h"
-#include <vector>
-#include <fstream>
-#include <algorithm>
-#include <winuser.h>
-#include "utils/WindowsUtils.h"
-#include <string>
-
-TraceLog::~TraceLog() {
-	// Nothing to do here, destructing is handled in FileLogBase
-}
-
-void TraceLog::writeJittedFunctionInfosToLog(std::vector<FunctionInfo>* functions)
-{
-	writeFunctionInfosToLog(LOG_KEY_JITTED, functions);
-}
-
-void TraceLog::writeInlinedFunctionInfosToLog(std::vector<FunctionInfo>* functions)
-{
-	writeFunctionInfosToLog(LOG_KEY_INLINED, functions);
-}
-
-void TraceLog::createLogFile(std::string targetDir) {
-
-	std::string timeStamp = getFormattedCurrentTime();
-
-	std::string fileName = "";
-	fileName = fileName + "coverage_" + timeStamp + ".txt";
-
-	FileLogBase::createLogFile(targetDir, fileName, true);
-
-	writeTupleToFile(LOG_KEY_INFO, VERSION_DESCRIPTION);
-	writeTupleToFile(LOG_KEY_STARTED, timeStamp.c_str());
-}
-
-void TraceLog::writeFunctionInfosToLog(const char* key, std::vector<FunctionInfo>* functions) {
-	for (std::vector<FunctionInfo>::iterator i = functions->begin(); i != functions->end(); i++) {
-		writeSingleFunctionInfoToLog(key, *i);
-	}
-}
-
-void TraceLog::writeSingleFunctionInfoToLog(const char* key, FunctionInfo& info) {
-	char signature[BUFFER_SIZE];
-	signature[0] = '\0';
-	sprintf_s(signature, "%i:%i", info.assemblyNumber,
-		info.functionToken);
-	writeTupleToFile(key, signature);
-}
-
+#include "TraceLog.h"
+#include "version.h"
+#include <vector>
+#include <fstream>
+#include <algorithm>
+#include <winuser.h>
+#include "utils/WindowsUtils.h"
+#include <string>
+#include <regex>
+#include <sstream>
+
+TraceLog::~TraceLog() {
+	// Nothing to do here, destructing is handled in FileLogBase
+}
+
+void TraceLog::writeJittedFunctionInfosToLog(std::vector<FunctionInfo>* functions)

```

Comment:
Generic: Is it intended to use a pointer here? I would assume that this suffices (as the vector is probably not changed)
```suggestion
void TraceLog::writeJittedFunctionInfosToLog(const std::vector<FunctionInfo>& functions)

```